cdap-common/src/main/java/io/cdap/cdap/common/metrics/MetricsReporterHook.java

```diff
@@ -102,6 +105,15 @@ public void postCall(HttpRequest request, HttpResponseStatus status, HandlerInfo
 
       // todo: report metrics broken down by status
       collector.increment("response." + name, 1/*, "status:" + code*/);
+
+      // store response time metric
+      long currTime = System.currentTimeMillis();
```

Comment:
@chtyim proposed to use `System.nanotime()` and do it in nanos. Makes sense for me as some calls can be pretty short.

Reply:
It maybe more convenient to keep it in milliseconds. We are probably not worried about API calls that take less than 1ms. If the metric is in nanos, every time we will find ourselves dividing by 1000 (or ignoring the last 3 digits). It will be more intuitive to use milliseconds I think. What do you think?

Reply:
Also currentTimeMillis() is affected by clock drift while nanoTime is generally monotonic nowadays. I think it is safer and more accurate to use nanoTime for measuring time elapse.

Reply:
Changed to using nanoTime